### ISSUE TITLE
Use some sort of linter for the documentation

### ISSUE THREAD

**Issue by arthurlutz** [home-assistant/home-assistant.github.io#8499]
It would be nice to have some sort of linter for the documentation to help existing and new contributors.

In the python/restructured text world, I'm used to using https://github.com/openstack/doc8

Does anyone know an nice equivalent for markdown ? 

For jekyll a quick search lead me to http://www.plainionist.net/Plainion.JekyllLint/ which seems to have a good approach but might be difficult to install/use (F#). There is probably some other projects.

Once this is in place, the linter could do checks on the metadata in the components documentation (check the taxonomies used there). (maybe another issue?)

**Comment by frenck** [comment]
Recently, the liquid checks have been tightened to fail on warnings.
I currently working on slowly making everything pass the Markdown linting.

Even though, in general, a good idea, we have to keep in mind that less technical users will contribute to the documentation. We should be careful not to scare those away since they often deliver to most valuable documentation (as seen from a user perspective).

**Comment by arthurlutz** [comment]
@frenck I am not suggesting linters should be mandatory for contributing. I'm just suggesting that it might be some help for certain quality assurance of the documentation. I've seen other projects where the maintainer team actually modifies the pull request to match quality standards to avoid making the project inaccessible to "non technical users". 

What tool are you using for "Markdown linting" ?

**Comment by frenck** [comment]
We do modify for "non technical users".
We simply focus on content for the docs, not how the Markdown looks.

Thanks.

**Comment by arthurlutz** [comment]
OK, I understand. We can close this issue if you want. Thanks for taking the time.

**Comment by frenck** [comment]
No problem! 👍
